components/neeva/renderer/content_filtering_agent.cc

```diff
@@ -53,6 +62,30 @@ std::unique_ptr<blink::URLLoaderThrottle> ContentFilteringAgent::CreateThrottle(
       base::WrapRefCounted(this), render_frame_id, request);
 }
 
+void ContentFilteringAgent::RunScriptsAtDocumentStart(
+    content::RenderFrame* render_frame) {
+  auto* web_frame = render_frame->GetWebFrame();
+  if (!web_frame)
+    return;
+
+  // Use the security origin here instead of the URL to cover script generated
+  // documents as well.
+  std::string host = web_frame->GetDocument().GetSecurityOrigin().Host().Utf8();
+
+  for (const auto& filter : filters_) {
+    if (!filter.css_matcher)
```

Comment:
more confirming my understanding then anything. GetStyleSheetForHost does a whole bunch of allocation when called, so ideally this if should return for any non-adblocker mode. So when adblocker is not enabled, we should not be creating a css_matcher. 

I wondered if there is a scenario where we still create one but with an empty/null rule_list_. If not the case no worries, but if that is how we are making the matcher work, then for all non-adblocker cases getstylesheetforhost will still be called, right?